pkg/kubectl/resource/visitor.go

```diff
@@ -430,6 +335,62 @@ func (v FlattenListVisitor) Visit(fn VisitorFunc) error {
 	})
 }
 
+func ignoreFile(path string, extensions []string) bool {
+	if len(extensions) == 0 {
+		return false
+	}
+	ext := filepath.Ext(path)
+	for _, s := range extensions {
+		if s == ext {
+			return false
+		}
+	}
+	return true
+}
+
+// ExpandPathsToStreamVisitors will return a slice of Stream Visitors that will read and handle files from the provided path.
+// The returned visitor will be a StreamVisitor as reading a file can be the same as reading from STDIN. Paths argument also
+// accepts a single file, and will return a single visitor
+func ExpandPathsToStreamVisitors(mapper *Mapper, paths string, recursive bool, extensions []string, ignoreErrors bool, schema validation.Schema) ([]Visitor, error) {
+	var visitors []Visitor
+
+	err := filepath.Walk(paths, func(path string, fi os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if fi.IsDir() {
+			if path != paths && !recursive {
+				return filepath.SkipDir
+			}
+			return nil
+		}
+		if ignoreFile(path, extensions) {
+			return nil
+		}
+
+		f, err := os.Open(path)
+		if err != nil {
+			return fmt.Errorf("unable to open %q: %v", path, err)
+		}
+
```

Comment:
I can use `ioutil.ReadFile` instead of `Open(path)`. That implies that we lose the generality of large files.

About leaking: If I document explicitly that this function is opening files then it is up to the user to close those files. It should be ok, no?

BTW, in the previous version we were already leaking STDIN. No one was closing it.

The way I see it is that we either specifically declare that `ExpandPathsToStreamVisitors` keeps files open until `StreamVisitor` visits them, or we read the whole file and keep it in memory from the beginning (we must also change the way we use STDIN in this case).